base/dates/types.jl

```diff
@@ -58,10 +58,11 @@ immutable Date <: TimeType
     Date(instant::UTInstant{Day}) = new(instant)
 end
 
-# Fallback constructors
-_c(x) = convert(Int64,x)
-DateTime(y,m=1,d=1,h=0,mi=0,s=0,ms=0) = DateTime(_c(y),_c(m),_c(d),_c(h),_c(mi),_c(s),_c(ms))
-Date(y,m=1,d=1) = Date(_c(y),_c(m),_c(d))
+# Time is a nanosecond precision representation of hours:minutes:seconds.fractional
+immutable Time <: TimeType
+    instant::Nanosecond
+    Time(instant::Nanosecond) = new(instant)
```

Comment:
Constructor is necessary, just like for the other TimeTypes. By default, every type is given two constructors, one with the exact arguments of its fields and another that allows "Any" with calls to convert of the field types. By defining just the one constructor according to the field type, we're *not* getting the "Any" constructor, which is desirable since we define our own later on (without this, we would get a method overwritten warning when we define our own "Any" fallback constructor)